WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadUtils.java

```diff
@@ -262,12 +262,20 @@ public static void publishPost(Activity activity, final PostModel post, SiteMode
         PostUtils.trackSavePostAnalytics(post, site);
     }
 
+    /*
+     * returns true if the user has permission to publish the post - assumed to be true for
+     * dot.org sites because we can't retrieve their capabilities
+     */
+    public static boolean userCanPublish(SiteModel site) {
```

Comment:
I guess we should use this in the post list, maybe there https://github.com/wordpress-mobile/WordPress-Android/blob/2790aa65e4b6cac76113ba3c7992c942e29e01fa/WordPress/src/main/java/org/wordpress/android/viewmodel/posts/PostListViewModel.kt#L360 since we can't trust `hasCapabilityPublishPosts` for self hosted sites.

Reply:
BTW I filled an issue about "Submit"/"Publish" button on the post list for self hosted https://github.com/wordpress-mobile/WordPress-Android/issues/10354 we should maybe do a follow up PR for that one.

Reply:
Good find! I'll fix both issues in the PR for the issue.

Reply:
Fixed in https://github.com/wordpress-mobile/WordPress-Android/pull/10356